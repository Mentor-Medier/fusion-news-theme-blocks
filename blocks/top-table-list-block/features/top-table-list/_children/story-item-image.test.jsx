import React from 'react';
import { shallow } from 'enzyme';
import StoryItemImage from './story-item-image';

describe('StoryItemImage', () => {
  const mockProps = {
    itemTitle: 'mocktitle',
    imageURL: '',
    websiteURL: 'https://mockUrl.com',
    arcSite: 'the-sun',
    resizedImageOptions: { '400x267': '' },
    targetFallbackImage: {}, // check console
    placeholderResizedImageOptions: {},
    imageRatio: {},
    element: {},
    customFields: {},
  };

  beforeAll(() => {
    jest.mock('fusion:properties', () => (jest.fn(() => ({
      fallbackImage: 'placeholder.jpg',
      resizerURL: 'http://example.com',
      primaryLogoAlt: '',
    }))));
    jest.mock('fusion:context', () => ({
      useFusionContext: jest.fn(() => ({
        arcSite: 'the-sun',
        globalContent: {},
      })),
    }));
  });

  afterAll(() => {
    jest.resetModules();
  });
  it('render placeholder image while imageUrl is empty', () => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    const wrapper = shallow(<StoryItemImage {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
