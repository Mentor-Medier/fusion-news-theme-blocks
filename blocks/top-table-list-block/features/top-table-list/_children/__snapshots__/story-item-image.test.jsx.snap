// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryItemImage render placeholder image while imageUrl is empty 1`] = `
<div
  className=""
>
  <div
    className="flex no-image-padding"
  >
    <div
      className="image-wrapper"
    >
      <Image
        alt="Placeholder logo"
        largeHeight={267}
        largeWidth={400}
        mediumHeight={183}
        mediumWidth={274}
        resizedImageOptions={Object {}}
        smallHeight={183}
        smallWidth={274}
        url={Object {}}
      />
      <PromoLabel
        size="small"
        type="other"
      />
    </div>
  </div>
</div>
`;
